### ISSUE TITLE
Remove dependency on ethereumjs-util

### ISSUE THREAD

**Issue by axic** [ethereumjs/merkle-patricia-tree#10]
It might make sense to remove the dependency on ethereumjs-util as it is only using sha3 and toBuffer.

Not entirely sure about it.

**Comment by kumavis** [comment]
relevant https://github.com/ethereumjs/ethereumjs-util/pull/38

**Comment by axic** [comment]
I think `toBuffer` shouldn't be used here as that is very permissive, while according to the code documentation, the trie only accepts `Buffer` or `String` for keys and values. I would go as far as saying it should only work with `Buffer` and reject anything else (as it is a low-level library).

As regarding to sha3, perhaps it should use keccakjs directly.

**Comment by wanderer** [comment]
yeah i'm happy limiting the api to only buffers.

**Comment by tuler** [comment]
If `ethereumjs-util` is not being removed I'd appreciate a new release (`3.1.0`?) to include the version update you already did [here](https://github.com/ethereumjs/merkle-patricia-tree/commit/3061de7c41e87190f55bcb5babdd74d8c2c8b6c7#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R53)

This is affecting [truffle build](https://github.com/trufflesuite/truffle/issues/1584)

**Comment by 4vadim4** [comment]
@tuler , what about solution ? i have same problem:

> npm ERR! keccak@1.4.0 rebuild: `node-gyp rebuild`
> npm ERR! Exit status 1
> npm ERR! 
> npm ERR! Failed at the keccak@1.4.0 rebuild script.

this is my dependency-tree:
```
├─┬ ganache-cli@6.7.0
│ └─┬ ethereumjs-util@6.1.0
│   └── keccak@1.4.0 
└─┬ web3@1.2.4
  └─┬ web3-eth@1.2.4
    └─┬ web3-eth-accounts@1.2.4
      └─┬ ethereumjs-tx@2.1.2
        └─┬ ethereumjs-util@6.2.0
          └── keccak@2.1.0 
```

**Comment by ryanio** [comment]
I believe we've reduced our usage of `ethereumjs-util` as much as makes sense. See https://github.com/ethereumjs/merkle-patricia-tree/pull/111#issuecomment-617346261. `toBuffer` is no longer used, and `keccak` is used appropriately.

 @4vadim4 The keccak dependency of `ethereumjs-util 7.0.0` was updated from 2.1.0 to 3.0.0, which comes with prebuilds for Linux, MacOS and Windows so hopefully you won't need to run node-gyp on installation anymore with mpt v4.0.0 (#111). cc @tuler 

I'll link the release to close this issue when merged.
